src/UI/TransactionsToastList/components/Toast/components/Transaction/index.tsx

```diff
@@ -0,0 +1,208 @@
+  const transactionDisplayInfo = useGetTransactionDisplayInfo(toastId);
+  const accountShard = useSelector(shardSelector);
+
+  const {
+    errorMessage = 'Transaction failed',
+    timedOutMessage = 'Transaction timed out',
+    successMessage = 'Transaction successful',
+    processingMessage = 'Processing transaction'
+  } = transactionDisplayInfo;
+
+  const currentTx: SignedTransactionsBodyType =
+    signedTransactionsToRender[toastId];
+
+  const { transactions, status } = currentTx;
+
+  const invalidCurrentTx =
+    currentTx?.transactions == null || currentTx?.status == null;
+
+  const isSameShard = useMemo(
+    () =>
+      transactions!.reduce(
+        (
+          prevTxIsSameShard: boolean,
+          { receiver, data }: SignedTransactionType
+        ) => {
+          const receiverAddress = getAddressFromDataField({
+            receiver,
+            data: data ?? ''
+          });
+          if (receiverAddress == null) {
+            return prevTxIsSameShard;
+          }
+          return (
+            prevTxIsSameShard &&
+            isCrossShardTransaction({
+              receiverAddress,
+              senderShard: accountShard
+            })
+          );
+        },
+        true
+      ),
+    [transactions, accountShard]
+  );
+
+  const shardAdjustedDuration = isSameShard
+    ? averageTxDurationMs
+    : crossShardRounds * averageTxDurationMs;
+
+  const transactionDuration =
+    transactionDisplayInfo?.transactionDuration || shardAdjustedDuration;
+
+  const [startTime, endTime] = useMemo(() => {
+    const startTime = startTimeProgress || moment().unix();
+    const endTime =
+      endTimeProgress ||
+      moment().add(Number(transactionDuration), 'milliseconds').unix();
+    return [startTime, endTime];
+  }, []);
+
+  if (invalidCurrentTx) {
+    return null;
+  }
+
+  if (transactions === null || !shouldRender) {
+    return null;
+  }
+
+  const progress = { startTime, endTime };
+  const style = getGeneratedClasses(className, shouldRenderDefaultCss, {
+    ...styles
+  });
+
+  const successToastData = {
+    id: toastId,
+    icon: icons.faCheck,
+    expires: 30000,
+    hasCloseButton: true,
+    title: successMessage,
+    iconClassName: style.success
+  };
+
+  const pendingToastData = {
+    id: toastId,
+    expires: false,
+    icon: icons.faHourglass,
+    hasCloseButton: false,
+    title: processingMessage,
+    iconClassName: style.warning
+  };
+
+  const failToastData = {
+    id: toastId,
+    icon: icons.faTimes,
+    title: errorMessage,
+    hasCloseButton: true,
+    iconClassName: style.danger
+  };
+
+  const timedOutToastData = {
+    id: toastId,
+    icon: icons.faTimes,
+    title: timedOutMessage,
+    hasCloseButton: true,
+    iconClassName: style.warning
+  };
+
+  const isPending = getIsTransactionPending(status);
+  const isTimedOut = getIsTransactionTimedOut(status);
+
+  const toatsOptionsData = {
+    signed: pendingToastData,
```

Comment:
What if we move this guy outside of the component, with the above constants, and make the values a function that receives the changeable variables (toastId, successMessage), so we clear out the component a bit?